viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/utils.tsx

```diff
@@ -12,17 +12,26 @@ const entryToTreeNode = (entry: Entry) => ({
 export const listToTree = (list: Entry[]): TreeNode[] => {
     const rootNodes: TreeNode[] = [];
     const lookup: { [key: string]: TreeNode } = {};
+    const noParentList: TreeNode[]  = [];
     list.forEach(entry => {
-        lookup[entry.id] = entryToTreeNode(entry);
+        const node = entryToTreeNode(entry);
+        lookup[entry.id] = node;
+        if (entry.parentId !== -1) {
+            const parent: TreeNode = lookup[entry.parentId];
+            if (parent) {
+                parent.children.push(node);
+            } else {
+                noParentList.push(node);
```

Comment:
So basically, if a child comes before its parent in the entry list, you add it to a list of entries with undefined parent and after all entries have been processed, you push those children at the end of the parent? If other children for that parent came later, then the child that first came will be after them in the list, that doesn't preserve the order of entries, no? Or am I mistaken?

Shouldn't you first prepare the lookup array, then fill the entries by taking the list order again?

Reply:
Ok, if a child comes before the parent then the order is not kept. I'll need to fix that.

I hope to get the order right without preparing the lookup array first which would avoid looping twice. I'll see what I can do.